src/Idris/Parser.idr

```diff
@@ -1766,7 +1766,7 @@ parameters {auto fname : OriginDesc} {auto indents : IndentInfo}
             cons   <- constraints fname indents
             n      <- decorate fname Typ name
             params <- many ifaceParam
-            det    <- option [] $ decoratedSymbol fname "|" *> sepBy (decoratedSymbol fname ",") (decorate fname Bound name)
+            det    <- optional $ decoratedSymbol fname "|" *> sepBy1 (decoratedSymbol fname ",") (decorate fname Bound name)
```

Comment:
This is where we could detect that we need at least one argument right? Maybe we could use this to provide a helpful error message in case we get `Nothing` and say we need at least one parameter

Reply:
Done